File: trademaster_ims/src/main/java/com/trademaster/ims/model/SaleItem.java
package com.trademaster.ims.model;

import jakarta.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name = "sales_items")
public class SaleItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "sales_item_id")
    private Long salesItemId;

    @Column(name = "sale_id", nullable = false)
    private Long saleId;

    @Column(name = "product_id", nullable = false)
    private Long productId;

    @Column(name = "quantity", nullable = false)
    private Integer quantity = 0;

    @Column(name = "unit_price", nullable = false, precision = 15, scale = 2)
    private BigDecimal unitPrice = BigDecimal.ZERO;

    @Column(name = "tax_rate", precision = 5, scale = 2)
    private BigDecimal taxRate = BigDecimal.ZERO;

    @Column(name = "discount_percent", precision = 5, scale = 2)
    private BigDecimal discountPercent = BigDecimal.ZERO;

    @Column(name = "discount_amount", precision = 15, scale = 2)
    private BigDecimal discountAmount = BigDecimal.ZERO;

    @Column(name = "total_price", nullable = false, precision = 15, scale = 2)
    private BigDecimal totalPrice = BigDecimal.ZERO;

    // Getters and setters
}

File: trademaster-ng/src/app/models/sale-item.ts

export interface ISaleItem {
  sales_item_id: number;
  sale_id: number;
  product_id: number;
  quantity: number;
  unit_price: number;
  tax_rate: number;
  discount_percent: number;
  discount_amount: number;
  total_price: number;
}

export class SaleItem implements ISaleItem {
  sales_item_id: number = 0;
  sale_id: number = 0;
  product_id: number = 0;
  quantity: number = 0;
  unit_price: number = 0;
  tax_rate: number = 0;
  discount_percent: number = 0;
  discount_amount: number = 0;
  total_price: number = 0;
}